### ISSUE TITLE
Feature: Add `DeepFilterCommandBuilder` to `MediaProcessor`

### ISSUE THREAD

**Issue by omeryusufyagci** [omeryusufyagci/fast-music-remover#32]
Implement a `DeepFilterCommandBuilder` class to establish a high-level API for constructing `DeepFilterNet` commands, extending the CommandBuilder base class. Methods such as `setInputAudio()` and `setNoiseReductionLevel()` should abstract the command details.

This will make the filtering process more maintainable and user-friendly, eliminating the need for users to understand command-level specifics. Will be needed by the frontend for v1.0.s

```cpp
DeepFilterCommandBuilder cmd;
cmd.setInputAudio(inputAudioPath);
cmd.setOutputAudio(outputAudioPath);
cmd.setNoiseReductionLevel(0.5);
// ...
```
See discussion in #4 for more context.

A good first issue as it's a direct extension of an existing class, requiring minimal understanding of the full project.

**Comment by georgeolson92** [comment]
Hey there! I wanted to see if you need some help getting this implemented. Would be happy to help!

**Comment by omeryusufyagci** [comment]
Hi @georgeolson92, it would be great, thanks a lot! Feel free to reach out if you have questions/feedback along the way.

**Comment by omeryusufyagci** [comment]
This was closed by #42
